packages/react-components/components/Button.v2.tsx

```diff
@@ -0,0 +1,133 @@
+import Touchable from '@celo/react-components/components/Touchable'
+import { colors } from '@celo/react-components/styles/colors.v2'
+import { fontStyles } from '@celo/react-components/styles/fonts.v2'
+import { debounce } from 'lodash'
+import React, { useCallback } from 'react'
+import { StyleProp, StyleSheet, Text, View, ViewStyle } from 'react-native'
+
+const BUTTON_TAP_DEBOUNCE_TIME = 300 // milliseconds
+const DEBOUNCE_OPTIONS = {
+  leading: true,
+  trailing: false,
+}
+
+export enum BtnTypes {
+  PRIMARY = 'Primary',
+  SECONDARY = 'Secondary',
+  TERTIARY = 'Tertiary',
+}
+
+export enum BtnSizes {
+  Small = 'small',
+  Medium = 'medium',
+  Full = 'full',
+}
+
+export interface ButtonProps {
+  onPress: () => void
+  style?: StyleProp<ViewStyle>
+  text: string | React.ReactNode
+  accessibilityLabel?: string
+  type: BtnTypes
+  disabled?: boolean
+  size?: BtnSizes
+  testID?: string
+  children?: React.ReactNode // remove? previously used for icons not sure if knew design will need but lets keep for now
```

Comment:
Yeah, open to removing til we need it. And when we need it I'd rather make a separate component and not let this one get bloated like the last one